### PR TITLE
Add support for cilium kube-proxy-replacement for redirect of requests

### PR DESCRIPTION
**Pull request by primeroz**
Towards https://github.com/giantswarm/giantswarm/issues/25360

### CODE REVIEW

File: helm/azure-ad-pod-identity-app/templates/localredirectpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{{- if and .Values.provider (eq .Values.provider "capz") -}}
```
Comment (whites11) [inline helm/azure-ad-pod-identity-app/templates/localredirectpolicy.yaml:1]: how about simply checking if the CRD exists?

Comment (primeroz) [reply helm/azure-ad-pod-identity-app/templates/localredirectpolicy.yaml:1]: I thought we did not want to install it on vintage, but if we do then sure i can do that :+1:

Comment (whites11) [reply helm/azure-ad-pod-identity-app/templates/localredirectpolicy.yaml:1]: I mean, if azure vintage will ever run cilium, then we need this so why not I guess

Comment (primeroz) [reply helm/azure-ad-pod-identity-app/templates/localredirectpolicy.yaml:1]: make sense, also means i don't have to update config repo :+1:

**Review by whites11 — APPROVED** [review]
daje

**Pull request merged**